### PR TITLE
Feature/save preferred sorting setting on marketplace generative

### LINKED ISSUE #554: Save preferred sorting setting on marketplace generative
mentioned by liquid:
> Unrelated, but is it possible to save the filtering used in a project marketplace page during a reload? 
> It always defaults to recently listed but that's rarely useful 😦 

In further discussion we figured we are referring to the sorting settings. 

Screenshot 2023-01-25 at 19 41 34

We should store the setting in  localstorage and also put it in the query params to allow url sharing

https://www.fxhash.xyz/marketplace/generative/24169

### PR DESCRIPTION
**Pull request by Jobayer12**
The PR fixes #554 
The save preferred sorting setting on marketplace generative has been implemented.

### CODE REVIEW

File: src/pages/marketplace/generative/[id].tsx [hunk 1]
```diff
@@ -317,6 +319,7 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
   return {
     props: {
       token: token,
+      urlQuery,
```
Comment (maerzhase) [inline src/pages/marketplace/generative/[id].tsx:322]: We should only return the query parameter we want to keep the seralised payload as small as possible

File: src/containers/Marketplace/GenerativeListings.tsx [hunk 2]
```diff
@@ -45,15 +47,49 @@ function sortValueToSortVariable(val: string) {
 
 interface Props {
   token: GenerativeToken
+  urlQuery: Record<string, string>
```
Comment (maerzhase) [inline src/containers/Marketplace/GenerativeListings.tsx:50]: We only need to provide the actualy initial sorting to the component

**Review by maerzhase — CHANGES_REQUESTED** [review]
First of all: Thanks a lot for the support @Jobayer12 
There are a few things that need to be adjusted. 

Sorry for not being more specific in the issue description but there is already a context that stores user settings and also handles the whole localStorage part. We should add the setting `prefereredMarketplaceSorting` there. See here:
 https://github.com/fxhash/fxhash-website/blob/a52cfa60ad522b8413c16b9ca9f8ab30160aae28/src/context/Theme.tsx#L13-L31
 
 I attached a few more code comments.

File: src/pages/marketplace/generative/[id].tsx (hunk 1 above)
Comment (Jobayer12) [reply src/pages/marketplace/generative/[id].tsx]: As per the direction of the comments, I made changes and push my code.

Questions:
1. I added "prefereredMarketplaceSorting" boolean field on ISettingsProperties interface. What to do next? Will it be functional later?

File: src/containers/Marketplace/GenerativeListings.tsx (hunk 2 above)
Comment (Jobayer12) [reply src/containers/Marketplace/GenerativeListings.tsx]: I made this change and please review my PR again.

File: src/pages/marketplace/generative/[id].tsx (hunk 1 above)
Comment (maerzhase) [reply src/pages/marketplace/generative/[id].tsx]: Sorry but I cant provide a step by step guidance. you have to checkout how the setting context works and store the setting there.

File: src/containers/Marketplace/GenerativeListings.tsx (hunk 2 above)
Comment (maerzhase) [reply src/containers/Marketplace/GenerativeListings.tsx]: What I meant was not to serialize the query object but only the value what we need. we dont have any other query params for now so we can literally return the sorting value. 

rename sort -> sorting

**Comment by maerzhase** [comment]
closing in favor of #575

**Pull request closed without merging**